File: memory-domain/src/main/java/com/memory/dto/member/MemberRequest.java
package com.memory.dto.member;

import com.memory.domain.member.Member;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.Getter;

public class MemberRequest {

    @Getter
    public static class Signup {
        @NotBlank(message = "이메일은 필수 입력값입니다.")
        @Email(message = "이메일 형식이 올바르지 않습니다.")
        private String email;

        @NotBlank(message = "비밀번호는 필수 입력값입니다.")
        private String password;

        @NotBlank(message = "이름은 필수 입력값입니다.")
        private String name;

        @NotBlank(message = "닉네임은 필수 입력값입니다.")
        private String nickname;

        public Signup(String email, String password, String name, String nickname) {
            this.email = email;
            this.password = password;
            this.name = name;
            this.nickname = nickname;
        }

        public Member toEntity(String encodedPassword) {
            return new Member(name, nickname, email, encodedPassword);
        }
    }

    @Getter
    public static class Login {
        @NotBlank(message = "이메일은 필수 입력값입니다.")
        @Email(message = "이메일 형식이 올바르지 않습니다.")
        private String email;

        @NotBlank(message = "비밀번호는 필수 입력값입니다.")
        private String password;

        public Login(String email, String password) {
            this.email = email;
            this.password = password;
        }
    }

    @Getter
    public static class Update {
        @NotBlank(message = "닉네임은 필수 입력값입니다.")
        private String nickname;

        @NotBlank(message = "파일 ID는 필수 입력값입니다.")
        private Long fileId;

        public Update(String nickname, Long fileId) {
            this.nickname = nickname;
            this.fileId = fileId;
        }
    }

    @Getter
    public static class PasswordUpdate {
        @NotBlank(message = "비밀번호는 필수 입력값입니다.")
        private String password;

        public PasswordUpdate(String password) {
            this.password = password;
        }
    }
}

File: memory-api/src/main/kotlin/com/memory/dto/member/MemberRequest.kt
package com.memory.dto.member

import com.memory.domain.member.Member
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

class MemberRequest {

    data class Signup(
        @field:NotBlank(message = "이메일은 필수 입력값입니다.")
        @field:Email(message = "이메일 형식이 올바르지 않습니다.")
        val email: String,

        @field:NotBlank(message = "비밀번호는 필수 입력값입니다.")
        val password: String,

        @field:NotBlank(message = "이름은 필수 입력값입니다.")
        val name: String,

        @field:NotBlank(message = "닉네임은 필수 입력값입니다.")
        val nickname: String
    ) {
        fun toEntity(encodedPassword: String): Member =
            Member(name, nickname, email, encodedPassword)
    }

    data class Login(
        @field:NotBlank(message = "이메일은 필수 입력값입니다.")
        @field:Email(message = "이메일 형식이 올바르지 않습니다.")
        val email: String,

        @field:NotBlank(message = "비밀번호는 필수 입력값입니다.")
        val password: String
    )

    data class Update(
        @field:NotBlank(message = "닉네임은 필수 입력값입니다.")
        val nickname: String,

        @field:NotNull(message = "파일 ID는 필수 입력값입니다.")
        val fileId: Long
    )

    data class PasswordUpdate(
        @field:NotBlank(message = "비밀번호는 필수 입력값입니다.")
        val password: String
    )
}